### PR TITLE
Fix state sync with UI

### PR DESCRIPTION
**Pull request by jpuri**
Fixes: #16314

PR fixes issue caused on Ui by state early sync. PR delays state sync to point when state has requires account and balance details.

### CODE REVIEW

**Comment by jpuri** [comment]
@darkwing : I am getting this error on service worker restart:

Screenshot 2022-11-07 at 2 46 47 PM

**Review by watchanan554 — APPROVED** [review]

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1090,21 +1090,15 @@ export default class MetamaskController extends EventEmitter {
...
-        this.submitPassword(password);
-      }
-      // Automatic login via storage encryption key
-      else if (isManifestV3) {
-        this.submitEncryptionKey();
-      }
+      this._loginUser();
+    } else {
+      this.emit('startUISync');
+      this.memStore.subscribe(this.sendUpdate.bind(this));
```
Comment (jpuri) [inline app/scripts/metamask-controller.js:1101]: `memStore` updates a lot as controllers initialise, delaying this `subscribe` helps to make sure that incomplete UI state is not flushed to UI.

File: app/scripts/ui.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ async function start() {
      * In case the UI is already rendered, only update the streams.
      */
     const messageListener = async (message) => {
-      if (message?.name === 'CONNECTION_READY') {
```
Comment (Gudahtt) [inline app/scripts/ui.js:81]: Is the old `CONNECTION_READY` message still used for anything? Maybe it can be removed if not.

**Review by Gudahtt — COMMENTED** [review]
This looks great! Good idea to delay state updates like that as well

File: app/scripts/ui.js (hunk 2 above)
Comment (watchanan554) [reply app/scripts/ui.js:81]: Hi

Comment (jpuri) [reply app/scripts/ui.js:81]: I removed it.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by darkwing — APPROVED** [review]
OUTSTANDING @jpuri !  This is excellent!

**Pull request merged**